#! /bin/sh /usr/share/dpatch/dpatch-run
## 06_genppdupdate_egrep_stdin_freeze.dpatch.dpatch by  <rleigh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: cups-genppdupdate should not freeze waiting on stdin when running egrep

@DPATCH@

diff -urN gutenprint-4.3.99+cvs20060521.original/src/cups/cups-genppdupdate.in gutenprint-4.3.99+cvs20060521/src/cups/cups-genppdupdate.in
--- gutenprint-4.3.99+cvs20060521.original/src/cups/cups-genppdupdate.in	2006-03-29 03:27:15.000000000 +0100
+++ gutenprint-4.3.99+cvs20060521/src/cups/cups-genppdupdate.in	2006-05-22 17:37:57.224645208 +0100
@@ -131,13 +131,15 @@
 	chomp;
 	push @ppd_files,  $_;
     }
-    open PPDFILES, '-|', 'egrep', '-i', '-L', 'Foomatic', @ppd_files or die "can't grep $ppdlist: $!";
-    @ppd_files = ();
-    while (<PPDFILES>) {
-	chomp;
-	push @ppd_files,  $_;
+    if (@ppd_files) {
+	open PPDFILES, '-|', 'egrep', '-i', '-L', 'Foomatic', @ppd_files or die "can't grep $ppdlist: $!";
+	@ppd_files = ();
+	while (<PPDFILES>) {
+	    chomp;
+	    push @ppd_files,  $_;
+	}
+	close PPDFILES or ($! == 0) or die "can't close grep pipe: $!";
     }
-    close PPDFILES or ($! == 0) or die "can't close grep pipe: $!";
 }
 
 
